lib/galaxy/datatypes/sequence.py

```diff
@@ -940,4 +946,84 @@ def sniff(self, filename):
                             pairs = True
                     if seq and coor and pairs:
                         return True
+        return False
+
+
+class DotBracket ( Sequence ):
+    edam_format = "format_1457"
+    file_ext = "dbn"
+    
+    sequence_regexp = re.compile( "^[ACGTURYKMSWBDHVN]+$", re.I)
+    structure_regexp = re.compile( "^[\(\)\.\[\]]*" )
+    
+    def set_meta( self, dataset, **kwd ):
+        """
+        Set the number of sequences and the number of data lines
+        in dataset.
+        """
+        if self.max_optional_metadata_filesize >= 0 and dataset.get_size() > self.max_optional_metadata_filesize:
+            dataset.metadata.data_lines = None
+            dataset.metadata.sequences = None
+            dataset.metadata.seconday_structures = None
+            return
+        
+        data_lines = 0
+        sequences = 0
+        
+        for line in file( dataset.file_name ):
+            line = line.strip()
+            data_lines += 1
+            
+            if line and line.startswith( '>' ):
+                sequences += 1
+        
+        dataset.metadata.data_lines = data_lines
+        dataset.metadata.sequences = sequences
+    
+    def sniff(self, filename):
+        """
+        The Dot-Bracket format is as follows:
+        
+        >sequenceName1
+        CCCaaaGGG
+        (((...)))
+        >sequenceName2
+        GGGuuuCCC
+        (((...)))
+        
+        Because it remains unclear whether the Dot-Bracket format may
+        contain multiple sequences per file, sniffing is only applied
+        on the first 3 lines.
+        """
+        
+        i = 0
+        
+        with open( filename, "r" ) as handle:
+            for line in handle:
+                line = line.strip()
+                
+                state = i % 3
```

Comment:
I realized this format has some unresolved 'issues'. I denoted them:

Comment lines
-------------
According to the following website:
http://rna.urmc.rochester.edu/Text/File_Formats.html

'The first line starts with a ">" character and a file title follows.'
This means that there is no support for comment lines.

EDAM has no clear specification:
http://edamontology.org/format_1457

At another website no details on this are given:
http://projects.binf.ku.dk/pgardner/bralibase/RNAformats.html


Multiple sequence
---

There is no definition given on multiple sequences. Similar to the comment lines,
this can be expected since the format seems to be derived from the FASTA format
which supports both comment lines as multiple sequences.


Pseudoknots
-----------
The major limitation of a dotbracket structure-string based only on parenthesis is
that it doesn't allow pseudoknots, as explained better at this website:

http://ultrastudio.org/en/Dot-Bracket_Notation

This can be circumvented by accepting more chars, whereas each pair of
chars allows one recursive level of pseudoknots extra. At the same
website it is suggested to use only the following two pairs: [ .. ] or
{ .. }. 

However, on the following website: http://projects.binf.ku.dk/pgardner/bralibase/RNAformats.html
examples are given for: [', ']', '{', '}', '<', '>', 'A', 'a', 'B', 'c', 'C', ...

The current implementation supports these chars: ( .. )   [ .. ].

I do not think the DotBracket definition in Galaxy should include all
possible chars because it's hacking around a limitation of this
format. For compatibility, it would be nice to also support { .. }


Free energy suffix
------------------
At the following website:

http://projects.binf.ku.dk/pgardner/bralibase/RNAformats.html

The definition adds the gibbs free energy as suffix. This suffix is not taken up in the definition as I implemented, and I don't think it's convenient to put the energy values in the structure line but rather in the header line or, if we decide to support them, in the comment lines.